utils/ioutil.go

```diff
@@ -55,29 +58,52 @@ func New() IoUtil {
 // Download downloads file by provided URL and places its content to provided destPath.
 // Returns error in a case of any problems.
 // Returns HTTPError if downloading is caused by non 2xx response from a service accessed by URL
-func (util *impl) Download(URL string, destPath string) error {
-	out, err := os.Create(destPath)
-	if err != nil {
-		return err
-	}
-	defer Close(out)
-
+func (util *impl) Download(URL string, destPath string, useContentDisposition bool) (string, error) {
 	resp, err := util.httpClient.Get(URL)
 	if err != nil {
-		return err
+		return "", err
 	}
 	defer Close(resp.Body)
 
 	if resp.StatusCode != http.StatusOK {
-		return NewHTTPError(resp, fmt.Sprintf("Downloading %s failed. Status code %v", URL, resp.StatusCode))
+		return "", NewHTTPError(resp, fmt.Sprintf("Downloading %s failed. Status code %v", URL, resp.StatusCode))
+	}
+
+	if useContentDisposition {
+		filePath, filename := filepath.Split(destPath)
+
+		if dispo := resp.Header.Get("Content-Disposition"); dispo != "" {
+			contentDispoFilename := ""
+			if _, params, err := mime.ParseMediaType(dispo); err == nil {
+				contentDispoFilename = params["filename"]
+			}
+			if strings.HasSuffix(contentDispoFilename, "/") || strings.Contains(contentDispoFilename, "\x00") {
+				contentDispoFilename = ""
+			}
+			contentDispoFilename = path.Base(path.Clean("/" + contentDispoFilename))
+			if contentDispoFilename == "." || contentDispoFilename == "/" {
+				contentDispoFilename = ""
+			}
```

Comment:
Could you add a few comments about why the steps above are necessary? Having multiple cases where `contentDispoFilename` is reset to the empty string is a little confusing. Some examples of cases being addressed would be useful.

Reply:
I refactored a bit the code to make it clearer in commit  https://github.com/eclipse/che-plugin-broker/pull/60/commits/103113f16e51cf895a242e69ec1659ebc3b82f53

Can you tell if you think a comment is still necessary ?